src/Aspire.Hosting/ApplicationModel/ResourceUpdatesAnnotation.cs

```diff
@@ -30,14 +30,58 @@ public sealed class CustomResourceAnnotation(Func<CustomResourceSnapshot> initia
     /// Updates the snapshot of the <see cref="CustomResourceSnapshot"/> for a resource.
     /// </summary>
     /// <param name="state">The new <see cref="CustomResourceSnapshot"/>.</param>
-    public async Task UpdateStateAsync(CustomResourceSnapshot state)
+    public Task UpdateStateAsync(CustomResourceSnapshot state)
```

Comment:
Are we leaving this as "Async" in case we want to do something async in the future?

Reply:
yep